pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/LawOfDemeterRule.java

```diff
@@ -301,7 +302,11 @@ private int variableDegree(ASTVariableAccess expr) {
 
         // note this max could be changed to min to get a more conservative
         // strategy, trading recall for precision. maybe make that configurable
-        return reaching.getReaching().stream().mapToInt(this::foreignDegree).max().orElse(TRUSTED);
+        return reaching.getReaching().stream()
+                // sort the assignments to have a deterministic result. We need to call #foreignDegree always
+                // in the same order, to get the same computed degrees.
+                .sorted(Comparator.comparing(AssignmentEntry::getLocation, Node.COORDS_COMPARATOR))
```

Comment:
```suggestion
                .sorted(Comparator.comparing(AssignmentEntry::getLocation, Node::compareLocation))
```
This comparator is more performant, essentially it's free for implementations of `AbstractJjtreeNode`, since it uses offsets directly instead of computing line/col numbers. We probably should remove `Node.COORDS_COMPARATOR` from the public API to avoid users/us making this mistake.

Reply:
:+1: